File: solutions/c/square-root/1/square_root.c
#include "square_root.h"

int square_root(int number) {
    if(number < 0) {
        return -1;
    }
    int result = 0;
    while(result*result < number) {
        result++;
    }
    if(number%result == 0) {
        return result;
    }
    else {
        return -1;
    }
}

File: solutions/julia/square-root/1/square-root.jl
function square_root(n)
    if n < 0 throw(DomainError("No real number returned.")) end
    if n <= 1 return n end
    max::Int128 = n ÷ 2
    min::Int128 = 1
    while max - min > 1
        bisect = (max-min) ÷ 2 + min
        estimate = bisect * bisect
        if estimate > n
            max = bisect
        elseif estimate < n
            min = bisect
        else
            return bisect
        end
    end
    if max * max > n return min end
    return max
end
